### PR TITLE
Automatic reload

### PR DESCRIPTION
**Pull request by varjolintu**
Tries reload automatically when KeePassXC is not started.

### CODE REVIEW

**Comment by Mogost** [comment]
Good feature!

File: keepassxc-browser/background/keepass.js [hunk 1]
```diff
@@ -996,3 +996,38 @@ keepass.decrypt = function(input, nonce) {
...
+        }
+    }, 1000);
+};
+
+keepass.reconnect = function(callback, tab) {
+    return new Promise((resolve, reject) => {
+        keepass.generateNewKeyPair();
+        keepass.changePublicKeys(tab).then((pkRes) => {
+            keepass.getDatabaseHash((gdRes) => {
+                if (gdRes) {
```
Comment (phoerious) [inline keepassxc-browser/background/keepass.js]: Negate and return. This gets deep enough already.

File: keepassxc-browser/background/keepass.js [hunk 2]
```diff
@@ -1005,3 +1005,41 @@ keepass.decrypt = function(input, nonce) {
...
+};
+
+keepass.reconnect = function(callback, tab) {
+    return new Promise((resolve, reject) => {
+        keepass.generateNewKeyPair();
+        keepass.changePublicKeys(tab).then((pkRes) => {
+            // Database hash should be received if the reconnection succeeded
+            keepass.getDatabaseHash((gdRes) => {
+                if (!gdRes) {
+                    reject(false);
```
Comment (phoerious) [inline keepassxc-browser/background/keepass.js]: Do we need a return after this?

Comment (phoerious) [reply keepassxc-browser/background/keepass.js]: Without a return, keepass.testAssociation will still be called after rejection and may produce ugly side effects.

Comment (varjolintu) [reply keepassxc-browser/background/keepass.js]: Debugged it and that indeed happens.. added the return.

Comment (phoerious) [reply keepassxc-browser/background/keepass.js]: You cannot change the state of a promise once it's been rejected or resolved, but the code after the call to reject() or resolve() is still being executed.

**Comment by Ealhad** [comment]
resolved #264

**Comment by varjolintu** [comment]
This can be re-merged after 2.3.4 is released. However, this means KeePassXC-Browser should set 2.3.4 as the minimum version.

**Comment by varjolintu** [comment]
Reminder for myself:
Disable reconnect if `navigator.platform` is `Win32` or `Windows` and KeePassXC version is lower than 2.3.4. With this there's no need to increase the minimum version. A mention to the documentation is needed. And maybe an additional error message to the extension.

**Pull request merged**